tests/std/tests/GH_000995_regex_custom_char_types/test.cpp

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// TRANSITION, GH-5563
+#pragma warning(push)
+#pragma warning(disable : 6510)
+#include <string>
+#pragma warning(pop)
+
+#include <cstddef>
+#include <regex>
+
+using namespace std;
+
+template <class UnderlyingChar, class FwdIt>
+basic_string<UnderlyingChar> convert_to_underlying_string(FwdIt first, FwdIt last) {
+    basic_string<UnderlyingChar> str;
+    for (; first != last; ++first) {
+        str.push_back(static_cast<UnderlyingChar>(*first));
+    }
+    return str;
+}
+
+template <class StringType, class UnderlyingChar>
+StringType convert_from_underlying_string(const basic_string<UnderlyingChar>& str) {
+    StringType result;
+    for (auto ch : str) {
+        result.push_back(static_cast<typename StringType::value_type>(ch));
+    }
+    return result;
+}
+
+template <class SourceChar, class UnderlyingChar>
+class test_regex_traits {
+private:
+    using rx_traits = regex_traits<UnderlyingChar>;
+
+public:
+    using char_type       = SourceChar;
+    using string_type     = basic_string<SourceChar>;
+    using locale_type     = typename rx_traits::locale_type;
+    using char_class_type = typename rx_traits::char_class_type;
+
+    test_regex_traits() = default;
+
+    static size_t length(const SourceChar* p) {
+        return string_type::char_traits::length(p);
+    }
+
+    SourceChar translate(const SourceChar c) const {
+        return c;
+    }
+
+    SourceChar translate_nocase(const SourceChar c) const {
+        return static_cast<SourceChar>(inner.translate_nocase(static_cast<UnderlyingChar>(c)));
+    }
+
+    template <class FwdIt>
+    string_type transform(FwdIt first, FwdIt last) const {
+        auto str = convert_to_underlying_string<UnderlyingChar>(first, last);
+        return convert_from_underlying_string<string_type>(inner.transform(str.begin(), str.end()));
+    }
+
+    template <class FwdIt>
+    string_type transform_primary(FwdIt first, FwdIt last) const {
+        auto str = convert_to_underlying_string<UnderlyingChar>(first, last);
+        return convert_from_underlying_string<string_type>(inner.transform(str.begin(), str.end()));

```

Comment:
It looks like a copy-paste mistake that `transform_primary` calls `inner.transform`. (If it was intended, it should be commented.)